source/MaterialXShaderGen/Syntax.h

```diff
@@ -12,95 +12,62 @@
 namespace MaterialX
 {
 
-class SgInput;
-class SgOutput;
-
 using SyntaxPtr = shared_ptr<class Syntax>;
+using TypeSyntaxPtr = shared_ptr<class TypeSyntax>;
 
-/// Base class for syntax objects used by code generators
-/// to emit code with correcy syntax for each language
+/// Base class for syntax objects used by shader generators
+/// to emit code with correcy syntax for each language.
 class Syntax
 {
 public:
-    /// Struct holding information on syntax to be used for data types
-    /// Required to be set for each data type
-    struct TypeSyntax
-    {
-        string name;                // type name
-        string defaultValue;        // default value
-        string paramDefaultValue;   // default value in a shader param initialization context
-        string typeDef;             // custom typedef if needed in source code
-        string outputName;          // type name in output context
-
-        TypeSyntax() {}
-        TypeSyntax(const string& n, const string& dv, const string& pdv, const string& dn, const string& on)
-            : name(n)
-            , defaultValue(dv)
-            , paramDefaultValue(pdv)
-            , typeDef(dn)
-            , outputName(on)
-        {}
-    };
-
-    /// Struct holding information on syntax needed to construct values and swizzle patterns for data types
-    /// Optional and only used if a custom constructor and/or swizzling is needed for a data type
-    struct ValueConstructSyntax
```

Comment:
Removed this and merged its value construction functionality into the TypeSyntax class